docker/install-dynamic-plugins.py

```
@@ -789,6 +792,87 @@ def wait_for_lock_release(lock_file_path):
+            max_entry_size = int(os.environ.get('MAX_ENTRY_SIZE', 20000000))
+
+            for layer in manifest.get('layers', []):
+                layer_digest = layer.get('digest', '')
+                if not layer_digest:
+                    continue
+
+                (_sha, filename) = layer_digest.split(':')
+                layer_file = os.path.join(local_dir, filename)
+                if not os.path.isfile(layer_file):
+                    print(f"\t==> WARNING: Layer file {filename} not found", flush=True)
+                    continue
+
+                print(f"\t==> Extracting layer {filename}", flush=True)
+                with tarfile.open(layer_file, 'r:*') as tar:
+                    for member in tar.getmembers():
+                        # Security checks
+                        if member.size > max_entry_size:
+                            print(f"\t==> WARNING: Skipping large file {member.name} in catalog index", flush=True)
+                            continue
+                        if member.islnk() or member.issym():
+                            realpath = os.path.realpath(os.path.join(catalog_index_temp_dir, *os.path.split(member.linkname)))
+                            if not realpath.startswith(catalog_index_temp_dir):
+                                print(f"\t==> WARNING: Skipping link outside archive: {member.name}", flush=True)
+                                continue
+                        tar.extract(member, path=catalog_index_temp_dir, filter='tar')
+
+        default_plugins_file = os.path.join(catalog_index_temp_dir, 'dynamic-plugins.default.yaml')
+        if os.path.isfile(default_plugins_file):
+            print(f"\t==> Successfully extracted catalog index with dynamic-plugins.default.yaml", flush=True)
+            return default_plugins_file
+        else:
+            print(f"\t==> Catalog index extracted but dynamic-plugins.default.yaml not found", flush=True)
+            return None
```

Same point: it should raise an exception if the expected `dynamic-plugins.default.yaml` could not be found.